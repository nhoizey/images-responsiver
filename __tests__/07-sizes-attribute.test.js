'use strict';

const prettier = require('prettier');
const imagesResponsiver = require('../index');

function cleanHtml(html) {
	return prettier.format(html, { parser: 'html' });
}

describe('sizes attribute', () => {
	test('existing sizes attribute is untouched', () => {
		const content = `<!DOCTYPE html><html><body>
      <img src="test1.png" sizes="50vw" /></body></html>`;
		const transformed = imagesResponsiver(content);

		expect(cleanHtml(transformed)).toMatchSnapshot();
	});

	test('existing sizes attribute is overriden', () => {
		const content = `<!DOCTYPE html><html><body>
      <img src="test1.png" sizes="50vw" /></body></html>`;
		const transformed = imagesResponsiver(content, {
			default: {
				sizesOverride: true,
			},
		});

		expect(cleanHtml(transformed)).toMatchSnapshot();
	});
});
