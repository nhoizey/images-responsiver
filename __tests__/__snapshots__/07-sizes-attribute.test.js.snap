// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sizes attribute existing sizes attribute is overriden 1`] = `
"<!DOCTYPE html>
<html>
  <body>
    <img
      data-pristine=\\"test1.png\\"
      srcset=\\"
        test1-320.png   320w,
        test1-560.png   560w,
        test1-800.png   800w,
        test1-1040.png 1040w,
        test1-1280.png 1280w
      \\"
      src=\\"test1-640.png\\"
      sizes=\\"100vw\\"
    />
  </body>
</html>
"
`;

exports[`sizes attribute existing sizes attribute is untouched 1`] = `
"<!DOCTYPE html>
<html>
  <body>
    <img
      data-pristine=\\"test1.png\\"
      srcset=\\"
        test1-320.png   320w,
        test1-560.png   560w,
        test1-800.png   800w,
        test1-1040.png 1040w,
        test1-1280.png 1280w
      \\"
      src=\\"test1-640.png\\"
      sizes=\\"50vw\\"
    />
  </body>
</html>
"
`;
